File: src/dataset/go/proverb_Exercism_go_drapala/gemini/GeminiFlash_proverb_v2.go
package proverb

import (
	"fmt"
	"strings"
)

func Proverb(rhyme []string) []string {
	length := len(rhyme)
	if length == 0 {
		return []string{}
	}

	proverb := make([]string, 0, length) // Pre-allocate slice for efficiency

	if length == 1 {
		proverb = append(proverb, fmt.Sprintf("And all for the want of a %s.", rhyme[0]))
		return proverb
	}

	for i := 0; i < length-1; i++ {
		proverb = append(proverb, fmt.Sprintf("For want of a %s the %s was lost.", rhyme[i], rhyme[i+1]))
	}

	proverb = append(proverb, fmt.Sprintf("And all for the want of a %s.", rhyme[0]))
	return proverb
}

func ProverbMaker(first, second string) string {
	var sb strings.Builder
	if second == "" {
		sb.WriteString("And all for the want of a ")
		sb.WriteString(first)
		sb.WriteString(".")
	} else {
		sb.WriteString("For want of a ")
		sb.WriteString(first)
		sb.WriteString(" the ")
		sb.WriteString(second)
		sb.WriteString(" was lost.")
	}
	return sb.String()
}

File: src/dataset/Java/proverb_exercism-java-blogscot/gemini/GeminiFlash_proverb_v2.java
class Proverb {
  private final String proverb;

  Proverb(String[] words) {
    if (words.length > 0) {
      StringBuilder builder = new StringBuilder();
      String previous = words[0];

      for (int i = 1; i < words.length; i++) {
        builder.append("For want of a ").append(previous).append(" the ").append(words[i]).append(" was lost.\n");
        previous = words[i];
      }

      builder.append("And all for the want of a ").append(words[0]).append(".");
      proverb = builder.toString();
    } else {
      proverb = "";
    }
  }

  String recite() {
    return proverb;
  }
}